rasterio/vfs.py

```diff
@@ -8,8 +8,19 @@
 # NB: As not to propagate fallacies of distributed computing, Rasterio
 # does not support HTTP or FTP URLs via GDAL's vsicurl handler. Only
 # the following local filesystem schemes are supported.
-SCHEMES = {'gzip': 'gzip', 'zip': 'zip', 'tar': 'tar', 'https': 'curl',
-           'http': 'curl', 's3': 's3'}
+SCHEMES = {
+    'gzip': 'gzip',
+    'gzip+file': 'gzip',
+    'zip': 'zip',
+    'zip+file': 'zip',
+    'tar': 'tar',
+    'tar+file': 'tar',
+    'https': 'curl',
+    'http': 'curl',
+    's3': 's3'}
+
+FILE_SCHEMES = [
+    '', 'file', 'gzip', 'gzip+file', 'zip', 'zip+file', 'tar', 'tar+file']
```

Comment:
TODO: add a test for a `zip+file://` dataset identifier. Not something we'll see much of, but it will pave the way for `zip+https://` and `zip+s3://`, which are going to be quite useful.

Reply:
I've made this a separate new issue: https://github.com/mapbox/rasterio/issues/1151.